fe/fe-core/src/main/java/com/starrocks/qe/scheduler/assignment/LocalFragmentAssignmentStrategy.java

```diff
@@ -190,9 +192,19 @@ private void assignScanRangesToColocateFragmentInstancePerWorker(
                     });
                 } else {
                     int expectedDop = Math.max(1, pipelineDop);
-                    List<List<Integer>> bucketSeqsPerDriverSeq = ListUtil.splitBySize(bucketSeqsOfInstance, expectedDop);
-
-                    instance.setPipelineDop(bucketSeqsPerDriverSeq.size());
+                    int expectedPhysicalDop = Math.min(expectedDop, bucketSeqsOfInstance.size());
+                    // For disable group execution logical dop == physical dop
+                    // For enable group execution logical dop >= physical dop
+                    int logicalDop = expectedPhysicalDop;
+                    if (fragment.isUseGroupExecution()) {
+                        // if fragment using group execution
+                        SessionVariable sv = ConnectContext.get().getSessionVariable();
+                        int maxDop = Math.min(sv.getGroupExecutionGroupScale() * expectedDop,
+                                sv.getGroupExecutionMaxGroups());
+                        logicalDop = Math.min(bucketSeqsOfInstance.size(), maxDop);
```

Comment:
Should we restrict `maxDop` not less than `expectedDop` like `maxDop = Math.max(maxDop, expectedDop)`? when users set `GroupExecutionGroupScale` or `GroupExecutionMaxGroups` to non-positive values by accidently.